app/models/broadcast_message.py

```diff
@@ -271,6 +272,26 @@ def add_areas(self, *new_area_ids):
         self.area_ids = list(OrderedSet(self.area_ids + list(new_area_ids)))
         self._update_areas()
 
+    def add_custom_areas(self, *circle_polygon, id):
+        if id not in self.area_ids:
+            simple_polygons = list(circle_polygon)
+            areas = {}
+            areas["ids"] = [id]
+            areas["names"] = [id]
+            areas["aggregate_names"] = []
+            if os.environ.get("IN_CICD"):
+                areas["simple_polygons"] = [
+                    [[round(coord, 10) for coord in polygon] for polygon in polygons] for polygons in simple_polygons
+                ]
```

Comment:
I think the triple-nested list comprehension here is at the limit of what's ok.  It's fine if the comprehension expression stays as simple as it is - any more complex than this and it'd probably be more readable if it was broken out into for-loops.